src/shared/components/HeaderTextField/HeaderTextField.tsx

```diff
@@ -0,0 +1,47 @@
+import React, { useRef, useEffect } from 'react'
+import { Container, WarningText, StyledInput } from './HeaderTextField.style'
+
+export type Variant = 'default' | 'error' | 'warning'
+
+export type HeaderTextFieldProps = {
+  value: string
+  warningText?: string
+  errorText?: string
+  onChange: (value: string) => void
+  variant?: Variant
+}
+type ChangeEvent = React.ChangeEvent<HTMLInputElement>
+
+const HeaderTextField = React.forwardRef<HTMLInputElement, HeaderTextFieldProps>(
+  ({ value, warningText, errorText, onChange, variant = 'default' }, ref) => {
+    const inputElement = useRef<HTMLInputElement>(null)
+    useEffect(() => {
+      if (inputElement.current === null) {
+        return
+      }
+      inputElement.current.style.width = value.length + 'ch'
```

Comment:
Like mentioned in the call, we cannot rely on the `value` as it will not be present in the uncontrolled state. Not sure how to solve this tbh